static/app/views/settings/project/sampling/modal/index.tsx

```diff
@@ -1,86 +1,234 @@
+              };
+            }
+            return {
+              category: name,
+              match: value.join('\n'),
+            };
+          }
+          return {category: name};
+        }),
+        sampleRate: sampleRate * 100,
+        errors: {},
+      };
+    }
+
+    return {
+      conditions: [],
+      sampleRate: null,
+      errors: {},
+    };
+  }
+
+  function getDescription() {
+    if (type === SamplingRuleType.TRACE) {
+      return {
+        title: rule ? t('Edit Distributed Trace Rule') : t('Add Distributed Trace Rule'),
+        description: tct(
+          'Using a Trace ID, select all Transactions distributed across multiple projects/services which match your conditions. However, if you only want to select Transactions from within this project, we recommend you add a [link] rule instead.',
+          {
+            link: (
+              <Link
+                to={recreateRoute(`${SamplingRuleType.TRANSACTION}/`, {
+                  routes,
+                  location,
+                  params,
+                  stepBack: -1,
+                })}
+              >
+                {t('Individual Transaction')}
+              </Link>
+            ),
+          }
+        ),
+      };
+    }
+
+    return {
+      title: rule
+        ? t('Edit Individual Transaction Rule')
+        : t('Add Individual Transaction Rule'),
+      description: tct(
+        'Select Transactions only within this project which match your conditions. However, If you want to select all Transactions distributed across multiple projects/services, we recommend you add a [link] rule instead.',
+        {
+          link: (
+            <Link
+              to={recreateRoute(`${SamplingRuleType.TRACE}/`, {
+                routes,
+                location,
+                params,
+                stepBack: -1,
+              })}
+            >
+              {t('Distributed Trace')}
+            </Link>
+          ),
+        }
+      ),
+    };
+  }
+
+  const {errors, conditions, sampleRate} = data;
+
+  function convertRequestErrorResponse(error: ReturnType<typeof getErrorMessage>) {
+    if (typeof error === 'string') {
+      addErrorMessage(error);
+      return;
+    }
+
+    switch (error.type) {
+      case 'sampleRate':
+        setData({...data, errors: {...errors, sampleRate: error.message}});
+        break;
+      default:
+        addErrorMessage(error.message);
+    }
   }
 
-  function handleSubmit({
-    sampleRate,
-    conditions,
-    submitRules,
-  }: Parameters<RuleModalProps['onSubmit']>[0]) {
+  async function handleSubmit() {
     if (!defined(sampleRate)) {
       return;
     }
 
     const newRule: SamplingRule = {
       // All new/updated rules must have id equal to 0
       id: 0,
-      type: tracing ? SamplingRuleType.TRACE : SamplingRuleType.TRANSACTION,
+      type:
+        type === SamplingRuleType.TRACE
+          ? SamplingRuleType.TRACE
+          : SamplingRuleType.TRANSACTION,
```

Comment:
Can we just pass `type` directly here? `type` isn't optional and it's of `SamplingRuleType` and it only has these two options so after removing the `tracing` variable this condition seems redundant now

Reply:
good catch!